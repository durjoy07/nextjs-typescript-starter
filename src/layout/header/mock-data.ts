import { IHeader } from './interface';

export const headerProps: IHeader = {
  logo: 'Lemon Hive',
  menuItems: [
    {
      label: 'Home',
      href: '/',
    },
    {
      label: 'About',
      href: '/',
    },
    {
      label: 'Service',
      href: '/',
    },
    {
      label: 'Blog',
      href: '/',
    },
  ],
};
