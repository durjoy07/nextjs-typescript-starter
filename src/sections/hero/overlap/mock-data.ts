import { IHeroWithOverlapContent } from './interface';

export const heroWithOverlapContentProps: IHeroWithOverlapContent = {
  title: 'Lemon Hive, We Build Lightning Fast Websites',
  description:
    'We Build Lightning Fast Websites With Amazing UX Using Headless Architectures',
  buttons: {
    buttonLeft: {
      label: 'Server Side Example',
      href: '/',
    },
    buttonRight: {
      label: 'Client Side Example',
      href: 'client-side',
    },
  },
};
