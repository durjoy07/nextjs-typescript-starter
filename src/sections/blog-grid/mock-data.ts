import { IBlogGrid } from './interface';
import { blogCardProps } from '@components/cards/blog/mock-data';
import { sectionDescriptionProps } from '@components/section-heading/mock-data';

export const blogGridProps: IBlogGrid = {
  sectionHeading: {
    title: 'Our Blog',
    description: sectionDescriptionProps,
  },
  blogs: [
    { ...blogCardProps, tags: ['Nextjs', 'Reactjs'] },
    { ...blogCardProps, tags: ['Nextjs', 'Reactjs'] },
    { ...blogCardProps, tags: ['Nextjs'] },
    { ...blogCardProps },
    { ...blogCardProps },
    { ...blogCardProps },
    { ...blogCardProps },
    { ...blogCardProps },
  ],
};
