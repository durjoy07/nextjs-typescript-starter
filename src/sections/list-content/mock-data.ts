import { sectionDescriptionProps } from '@components/section-heading/mock-data';
import { ISanityCustomPortableText } from '@root/src/backend/types/portable-text';
import { IListContentSection } from './interface';

export const descriptionPortableData: ISanityCustomPortableText = [
  {
    _key: 'd956675f69e0',
    children: [
      {
        _type: 'span',
        marks: [],
        text: 'For design, branding, web development, marketing, consulting & other agencies.',
        _key: '612b8d7ad65a0',
      },
    ],
    markDefs: [],
    _type: 'customBlock',
    style: 'normal',
  },
  {
    _key: 'f9a72086f01c',
    children: [
      {
        _type: 'span',
        marks: [],
        text: 'We work with agencies in a number of different ways, but predominantly as an extension of your team to add headless development expertise.',
        _key: '3ec390abb3560',
      },
    ],
    markDefs: [],
    _type: 'customBlock',
    style: 'normal',
  },
  {
    _key: 'af14faef2a19',
    children: [
      {
        _type: 'span',
        marks: [],
        text: 'With the move towards headless website development, it can be difficult to find the right people, with the right expertise, to produce websites to your standards. By outsourcing the development and/or design to us, we can ensure you build exceptional headless websites that exceed what is possible with monolithic platforms such as WordPress.',
        _key: '57de12dc89fc0',
      },
    ],
    markDefs: [],
    _type: 'customBlock',
    style: 'normal',
  },
];

export const listContentSectionProps: IListContentSection = {
  sectionTheme: 'secondary',
  sectionHeading: {
    title: 'Who We Work With',
    description: sectionDescriptionProps,
  },
  lists: [
    {
      title: 'Services for Agencies',
      description: descriptionPortableData,
    },
    {
      title: 'Services for Agencies',
      description: descriptionPortableData,
    },
  ],
};
