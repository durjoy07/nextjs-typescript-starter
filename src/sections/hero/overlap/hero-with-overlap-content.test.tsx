import { render } from '@testing-library/react';
import HeroWithOverlapContent from '.';
import { heroWithOverlapContentProps } from './mock-data';

it('renders hero with overlp content correctly', () => {
  const { getAllByRole } = render(
    <HeroWithOverlapContent {...heroWithOverlapContentProps} />
  );
  const element = getAllByRole('presentation')[0];

  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});
