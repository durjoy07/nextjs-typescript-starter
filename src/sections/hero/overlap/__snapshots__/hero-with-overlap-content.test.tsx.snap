// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders hero with overlp content correctly 1`] = `
<section
  role="presentation"
>
  <div
    class=" flex min-h-[500px] justify-between bg-black lg:min-h-[700px]"
  >
    <div
      class="container"
      role="presentation"
    >
      <div
        class="max-w-[1200px] content-wrapper"
        style="--overlap-content-height-half: 0px;"
      >
        <div
          class="space-y-12"
        >
          <h2
            class="text-3xl text-white md:text-4xl 2xl:text-5xl"
          >
            Lemon Hive, We Build Lightning Fast Websites
          </h2>
          <p
            class="text-xl text-white"
          >
            We Build Lightning Fast Websites With Amazing UX Using Headless Architectures
          </p>
          <div
            class="inline-flex flex-wrap items-center gap-4"
          >
            <a
              class="inline-flex focus:outline-none focus-visible:outline-none focus:shadow-none focus-visible:shadow-none"
              href="/"
              target="_self"
            >
              <span
                aria-label="Server Side Example"
                class="relative z-10 group/button overflow-hidden inline-flex items-center justify-center gap-3 cursor-pointer text-white hover:text-white leading-[1.15] button primary-bg-text-white hover-secondary-bg-text-white sm"
              >
                <span>
                  Server Side Example
                </span>
                <span
                  class="absolute z-[-1] inset-0 invisible opacity-0 group-hover/button:visible group-hover/button:opacity-100 button-bg"
                />
              </span>
            </a>
            <a
              class="inline-flex focus:outline-none focus-visible:outline-none focus:shadow-none focus-visible:shadow-none"
              href="client-side"
              target="_self"
            >
              <span
                aria-label="Client Side Example"
                class="relative z-10 group/button overflow-hidden inline-flex items-center justify-center gap-3 cursor-pointer text-white hover:text-white leading-[1.15] button secondary-bg-text-white hover-primary-bg-text-white sm"
              >
                <span>
                  Client Side Example
                </span>
                <span
                  class="absolute z-[-1] inset-0 invisible opacity-0 group-hover/button:visible group-hover/button:opacity-100 button-bg"
                />
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="-translate-y-1/2"
    style="margin-bottom: -0px;"
  >
    <div
      class="container"
      role="presentation"
    >
      <div
        class="rounded-10 bg-[#ddd] px-12 py-10 shadow-1"
      >
        <div
          class="space-y-6"
        >
          <h2
            class="text-3xl text-black md:text-4xl"
          >
            We Build Lightning Fast Websites With Amazing UX Using Headless
          </h2>
          <p
            class="text-xl text-black"
          >
            We make secure, flexible and fast headless (Jamstack) websites and apps that improve Technical SEO, increase visitor numbers & conversion rates. We build with Sanity, BigCommerce, Prismic, Strapi, Contentful, WordPress, and other headless content management systems.
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
